// @phntms/react-share/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocialPlatforms = exports.getCurrentUrlAndCopyToClipboard = exports.getShareUrl = exports.getWhatsAppUrl = exports.getFacebookUrl = exports.getTwitterUrl = exports.getLinkedinUrl = exports.MetaHeadEmbed = void 0;
var MetaHeadEmbed_1 = require("./components/MetaHeadEmbed");
Object.defineProperty(exports, "MetaHeadEmbed", { enumerable: true, get: function () { return __importDefault(MetaHeadEmbed_1).default; } });
var getLinkedinUrl_1 = require("./utils/getLinkedinUrl");
Object.defineProperty(exports, "getLinkedinUrl", { enumerable: true, get: function () { return __importDefault(getLinkedinUrl_1).default; } });
var getTwitterUrl_1 = require("./utils/getTwitterUrl");
Object.defineProperty(exports, "getTwitterUrl", { enumerable: true, get: function () { return __importDefault(getTwitterUrl_1).default; } });
var getFacebookUrl_1 = require("./utils/getFacebookUrl");
Object.defineProperty(exports, "getFacebookUrl", { enumerable: true, get: function () { return __importDefault(getFacebookUrl_1).default; } });
var getWhatsAppUrl_1 = require("./utils/getWhatsAppUrl");
Object.defineProperty(exports, "getWhatsAppUrl", { enumerable: true, get: function () { return __importDefault(getWhatsAppUrl_1).default; } });
var getShareUrl_1 = require("./utils/getShareUrl");
Object.defineProperty(exports, "getShareUrl", { enumerable: true, get: function () { return __importDefault(getShareUrl_1).default; } });
var getCurrentUrlAndCopyToClipboard_1 = require("./utils/getCurrentUrlAndCopyToClipboard");
Object.defineProperty(exports, "getCurrentUrlAndCopyToClipboard", { enumerable: true, get: function () { return __importDefault(getCurrentUrlAndCopyToClipboard_1).default; } });
var types_1 = require("./types");
Object.defineProperty(exports, "SocialPlatforms", { enumerable: true, get: function () { return types_1.SocialPlatforms; } });
